### ISSUE TITLE
Seeking advice from people with a lot of experience learning AI

### ISSUE THREAD

**Issue by openmll** [ultralytics/ultralytics#20394]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hello, I am studying detection.
I am reimplementing YOLO for study and implementing the learning process simply. I also reconfigured the learning data loader. YOLO learned COCO with the yolon pretrain model.
The problem is that when learning is performed, an appropriate box is found in the initial epoch with confidence 10~50%. However, even though learning progresses to 100 epochs, each confidence converges to around confidence of 10%. I confirmed that the loss is decreasing overall.
A special point is that SGD is advantageous for learning and the smaller the batch size, the better (currently in progress with batch size=4).
Is there an error that those with a lot of experience in detection tasks like Yolo or AI can infer from this phenomenon?

thank you.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @openmll, thank you for your interest in Ultralytics 🚀! This is an automated response to help guide you while an Ultralytics engineer reviews your question.

We recommend checking out the [Docs](https://docs.ultralytics.com/) for a broad range of [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples, which cover many common scenarios and troubleshooting tips.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us investigate further.

For custom training or implementation ❓ questions like yours, please share as much detail as possible—including code snippets, dataset information, and training logs. Make sure you’re following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community for support and discussion:
- [Discord](https://discord.com/invite/ultralytics) for real-time chat 🎧
- [Discourse](https://community.ultralytics.com/) for in-depth discussions
- [Subreddit](https://reddit.com/r/Ultralytics) to connect and share with peers

## Upgrade

Upgrade to the latest `ultralytics` package, including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml), in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to ensure you are using the most up-to-date features and fixes:

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of the following verified environments with all dependencies (including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/)) preinstalled:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

An Ultralytics engineer will review your question and provide further assistance soon. Thank you for contributing to the Ultralytics community! 🦾

**Comment by glenn-jocher** [comment]
This sounds like a classic example of model overconfidence correction during training. As the model learns to better localize objects, it often becomes more "honest" about its uncertainty, leading to lower confidence scores despite improving overall performance. 

A few suggestions:

1. Check your confidence metric against mAP metrics - decreasing confidence with improving mAP is actually normal and desirable
2. Try adjusting your learning rate schedule - perhaps use warmup_epochs=0 as mentioned in our docs to prevent learning rate from starting too low
3. Evaluate your model on validation data throughout training to ensure performance is actually improving
4. Consider using mixed precision training to benefit from larger batch sizes while maintaining memory efficiency

SGD with smaller batches often works well for YOLO models, so your approach is reasonable. The key question is whether your model's actual detection performance (precision/recall) is improving, regardless of the confidence values.

**Comment by openmll** [comment]
> This sounds like a classic example of model overconfidence correction during training. As the model learns to better localize objects, it often becomes more "honest" about its uncertainty, leading to lower confidence scores despite improving overall performance.
> 
> A few suggestions:
> 
> 1. Check your confidence metric against mAP metrics - decreasing confidence with improving mAP is actually normal and desirable
> 2. Try adjusting your learning rate schedule - perhaps use warmup_epochs=0 as mentioned in our docs to prevent learning rate from starting too low
> 3. Evaluate your model on validation data throughout training to ensure performance is actually improving
> 4. Consider using mixed precision training to benefit from larger batch sizes while maintaining memory efficiency
> 
> SGD with smaller batches often works well for YOLO models, so your approach is reasonable. The key question is whether your model's actual detection performance (precision/recall) is improving, regardless of the confidence values.



Thanks for the advice on overconfidence and SGD. The key issue I want to know is that even though I increased the weight of the classification loss, the model keeps learning to predict lower confidence. Also, it is strange that the learning is not common even though it is common to use the basic YOLO configuration and the COCO dataset of 40,000 images.

**Comment by glenn-jocher** [comment]
This behavior of decreasing confidence despite increasing classification loss weight is actually a common phenomenon in object detection training. What's happening is that your model is likely becoming better calibrated - learning to be more conservative with its predictions rather than overconfident.

Try using the rect=True parameter during training, which can help with handling image sizes more effectively. Also, look at your learning rate - the default schedule might be dropping too quickly for your implementation. 

Most importantly, track your precision-recall curves rather than just confidence values. If your model is detecting objects correctly but with lower confidence, that's often preferable to high confidence but poor localization. Consider evaluating with lower confidence thresholds (0.1 or 0.05) during validation to see if performance is actually improving despite the confidence decrease.
